cmd/disk-cache.go

```diff
@@ -185,12 +185,17 @@ func (c *cacheObjects) GetObjectNInfo(ctx context.Context, bucket, object string
 	cacheReader, cacheErr := c.get(ctx, dcache, bucket, object, rs, h, opts)
 	if cacheErr == nil {
 		cc = cacheControlOpts(cacheReader.ObjInfo)
-		if !cc.isStale(cacheReader.ObjInfo.ModTime) {
+		if (!cc.isEmpty() && !cc.isStale(cacheReader.ObjInfo.ModTime)) ||
```

Comment:
why are you bringing the cc.isEmpty() back? because of cc.isStale returns false if cc.isEmpty() is true -  if cc.isEmpty() is true and cc.isEmpty() returns true otherwise? 

they both will negate the logic and it will never try to cache? - this was the bug that was fixed by @nitisht AFAIK

Reply:
This if condition is meant to handle only the case where cache-control header is set - to return cached entry without a HEAD operation to validate the ETag. After ETag validation, content is still being served from cache when cc.isEmpty() holds true.

Reply:
Understood